apps/extension/src/ui/domains/Settings/ManageNetworks/NetworkForm/SubNetworkForm.tsx

```diff
@@ -0,0 +1,391 @@
+    formState: { errors, isValid, isSubmitting, isDirty, touchedFields },
+  } = formProps
+
+  const { rpcs, id, nativeTokenCoingeckoId, accountFormat } = watch()
+
+  // initialize form with existing values (edit mode), only once
+  const initialized = useRef(false)
+  useEffect(() => {
+    if (chainId && defaultValues && !initialized.current) {
+      reset(defaultValues)
+      trigger("rpcs")
+      initialized.current = true
+    }
+  }, [defaultValues, chainId, reset, trigger])
+
+  // auto detect genesis hash (and token deets) based on RPC url (add mode only)
+  const rpcInfo = useRpcInfo(rpcs?.[0]?.url)
+  useEffect(() => {
+    if (chainId || !rpcInfo.isFetched) return
+    const rpcId = rpcInfo.data && `custom-${rpcInfo.data.genesisHash}`
+    if (rpcInfo.data && rpcId && rpcId !== id) {
+      setValue("id", rpcId, { shouldValidate: true, shouldTouch: true, shouldDirty: true })
+      setValue("genesisHash", rpcInfo.data.genesisHash as `0x${string}`, {
+        shouldValidate: true,
+        shouldTouch: true,
+        shouldDirty: true,
+      })
+      setValue("nativeTokenSymbol", rpcInfo.data.token.symbol, {
+        shouldValidate: true,
+        shouldTouch: true,
+        shouldDirty: true,
+      })
+      setValue("nativeTokenDecimals", rpcInfo.data.token.decimals, {
+        shouldValidate: true,
+        shouldTouch: true,
+        shouldDirty: true,
+      })
+    } else if (!!id && !rpcInfo.data) {
+      resetField("id")
+      resetField("genesisHash")
+      resetField("nativeTokenSymbol")
+      resetField("nativeTokenDecimals")
+    }
+  }, [id, resetField, setValue, chainId, rpcInfo.isFetched, rpcInfo.data])
+
+  // fetch token logo's url, but only if form has been edited to reduce 429 errors from coingecko
+  const coingeckoLogoUrl = useCoinGeckoTokenImageUrl(isDirty ? nativeTokenCoingeckoId : null)
+
+  const nativeTokenLogoUrl = useMemo(
+    // existing icon has priority
+    () =>
+      touchedFields?.nativeTokenCoingeckoId
+        ? coingeckoLogoUrl
+        : defaultValues?.nativeTokenLogoUrl ?? coingeckoLogoUrl,
+    [coingeckoLogoUrl, defaultValues?.nativeTokenLogoUrl, touchedFields?.nativeTokenCoingeckoId]
+  )
+
+  const chainLogoUrl = useMemo(
+    () => defaultValues?.chainLogoUrl ?? null,
+    [defaultValues?.chainLogoUrl]
+  )
+
+  useEffect(() => {
+    // check only if adding a new network
+    if (chainId || !id) return
+
+    if (chains?.some((c) => c.id === id)) {
+      if (!errors.id) setError("id", { message: t("already exists") })
+    } else {
+      if (errors.id) clearErrors("id")
+    }
+  }, [clearErrors, id, setError, errors.id, t, chainId, chains])
+
+  const accountFormatOptions = useMemo(
+    () => [
+      {
+        label: (
+          <div className="flex flex-col">
+            <div className="text-xs">{t("Polkadot (default)")}</div>
+            <div className="text-body-disabled text-tiny">
+              {shortenAddress("5C4hrfjw9DjXZTzV3MwzrrAr9P1MJhSrvWGWqi1eSuyUpnhM")}
```

Comment:
might want to use Alice address here, looks like this one belongs to someone

Reply:
Oh, I thought this was supposed to be like the null address `0x000...` in ethereum

Reply:
Hah, a fun read: https://pacna.org/dot-zero-addr/

Reply:
What's your preference from these two @0xKheops ?

Left: Alice
Right: Null/Zero

<img width="1397" alt="Screenshot 2023-09-13 at 14 40 03" src="https://github.com/TalismanSociety/talisman/assets/2741569/6f33c38f-d288-4e40-9325-ec59eba052d0">

Reply:
ok sorry about that, didn't know about this null/zero address thing.
as long as it's not the address of a random dude, all good :)

Awesome read too :joy: